packages/async-queue/src/lib/AsyncQueue.ts

```diff
@@ -12,21 +14,21 @@ export class AsyncQueue {
 	/**
 	 * The promises array
 	 */
-	private promises: InternalAsyncQueueDeferredPromise[] = [];
+	private promises: AsyncQueueEntry[] = [];
 
 	/**
 	 * Waits for last promise and queues a new one
 	 * @example
 	 * ```typescript
 	 * const queue = new AsyncQueue();
 	 * async function request(url, options) {
-	 *     await queue.wait();
+	 *     await queue.wait({ signal: options.signal });
 	 *     try {
 	 *         const result = await fetch(url, options);
 	 *         // Do some operations with 'result'
 	 *     } finally {
 	 *         // Remove first entry from the queue and resolve for the next entry
-	 *         queue.shift();
+	 *         queue.unlock();
```

Comment:
Forgotten rename?

Reply:
Yes - it contained the new code but returned a boolean: whether or not a new item is run. And `shift` would be deprecated calling it.

But then I remembered that AQ is supported in browsers, which drops the possibility to use Node's deprecate utilities, and that I would need to add a test testing a deprecated API or else coverage drops below 100%.

It was just better to leave it as is and keep the PR's scope as small as possible.